File: app/src/main/java/com/suhang/networkmvp/widget/autolayout/ChatLayout.java
package com.suhang.networkmvp.widget.autolayout;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

/**
 * Created by sh on 2016/6/28 20:51.
 */

public class ChatLayout extends AutoHeightLayout {
	private int softKeyHeight;
	private View controller;
	public ChatLayout(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	public void setController(View controller) {
		this.controller = controller;
	}


	@Override
	public void onSoftKeyboardHeightChanged(int height) {
		softKeyHeight = height;
	}

	public int getSoftKeyHeight() {
		return softKeyHeight;
	}

	@Override
	public void OnSoftPop(int height) {
		super.OnSoftPop(height);
		if (onSoftKeyStatuListener != null) {
			onSoftKeyStatuListener.onSoftKeyOpen(height);
		}
	}

	@Override
	public void OnSoftClose() {
		super.OnSoftClose();
		if (onSoftKeyStatuListener != null) {
			onSoftKeyStatuListener.onSoftKeyClose();
		}
	}

	private OnSoftKeyStatuListener onSoftKeyStatuListener;

	public void setOnSoftKeyHeightListener(OnSoftKeyStatuListener onSoftKeyStatuListener) {
		this.onSoftKeyStatuListener = onSoftKeyStatuListener;
	}
	public interface OnSoftKeyStatuListener {
		void onSoftKeyOpen(int height);
		void onSoftKeyClose();
	}
}

File: app/src/main/java/com/suhang/networkmvp/widget/autolayout/ChatLayout.kt
package com.suhang.networkmvp.widget.autolayout

import android.content.Context
import android.util.AttributeSet
import android.view.View

/**
 * Created by sh on 2016/6/28 20:51.
 */

class ChatLayout(context: Context, attrs: AttributeSet) : AutoHeightLayout(context, attrs) {
    var softKeyHeight: Int = 0
        private set
    private var controller: View? = null

    fun setController(controller: View) {
        this.controller = controller
    }


    override fun onSoftKeyboardHeightChanged(height: Int) {
        softKeyHeight = height
    }

    override fun OnSoftPop(height: Int) {
        super.OnSoftPop(height)
        if (onSoftKeyStatuListener != null) {
            onSoftKeyStatuListener!!.onSoftKeyOpen(height)
        }
    }

    override fun OnSoftClose() {
        super.OnSoftClose()
        if (onSoftKeyStatuListener != null) {
            onSoftKeyStatuListener!!.onSoftKeyClose()
        }
    }

    private var onSoftKeyStatuListener: OnSoftKeyStatuListener? = null

    fun setOnSoftKeyHeightListener(onSoftKeyStatuListener: OnSoftKeyStatuListener) {
        this.onSoftKeyStatuListener = onSoftKeyStatuListener
    }

    interface OnSoftKeyStatuListener {
        fun onSoftKeyOpen(height: Int)
        fun onSoftKeyClose()
    }
}
